sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -2095,10 +2095,12 @@ case class ArrayPosition(left: Expression, right: Expression)
 case class ElementAt(
     left: Expression,
     right: Expression,
+    // The value to return if index is out of bound
+    defaultValueOutOfBound: Expression = null,
```

Comment:
the default value for this parameter should be `Literal(null, NullType)`

Reply:
The type can be `Literal` to be more type-safe

Reply:
Actually, it's better to use `Option[Literal]`, where None means no default value and we should keep the old code path.